### PR TITLE
Migrate Tables extension code

### PR DESCRIPTION
**Pull request by pakrym**
Moves the code with as few changes as possible.

Copies some code from WebJobs repo to get things building.

Some functional tests were using storage queues as the trigger. I disabled those.

### CODE REVIEW

File: eng/Directory.Build.Common.props [hunk 1]
```diff
@@ -100,12 +100,6 @@
     </NoWarn>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(InheritDocEnabled)' != 'false' and '$(TargetFramework)' == 'netstandard2.0'">
```
Comment (pakrym) [inline eng/Directory.Build.Common.props:103]: TargetFramework is only available after csproj evaluation so should be consumed in targets.

File: sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/AzureStorageWebJobsStartup.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+using Microsoft.Azure.WebJobs.Extensions.Storage;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Extensions.Hosting;
+
+[assembly: WebJobsStartup(typeof(AzureStorageWebJobsStartup))]
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage
+{
+    public class AzureStorageWebJobsStartup : IWebJobsStartup
```
Comment (JoshLove-msft) [inline sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/AzureStorageWebJobsStartup.cs:12]: Don't we use this type in the Storage extensions?

**Comment by JoshLove-msft** [comment]
Is the idea here that we are trying to merge in what would be the T1 version of the extension but using the package naming pattern established in T2?

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/AzureStorageWebJobsStartup.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Extensions.Storage;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Extensions.Hosting;
+
+[assembly: WebJobsStartup(typeof(AzureStorageWebJobsStartup))]
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage
```
Comment (christothes) [inline sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/AzureStorageWebJobsStartup.cs:10]:
```suggestion
namespace Microsoft.Azure.WebJobs.Extensions.Tables
```

File: sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/StorageWebJobsBuilderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Bindings.StorageAccount;
+using Microsoft.Azure.WebJobs.Host.Tables.Config;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+namespace Microsoft.Extensions.Hosting
+{
+    public static class StorageWebJobsBuilderExtensions
+    {
+        public static IWebJobsBuilder AddAzureStorage(this IWebJobsBuilder builder)
```
Comment (christothes) [inline sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/StorageWebJobsBuilderExtensions.cs:14]:
```suggestion
        public static IWebJobsBuilder AddAzureTables(this IWebJobsBuilder builder)
```

File: sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/StorageWebJobsBuilderExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Storage;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Bindings.StorageAccount;
+using Microsoft.Azure.WebJobs.Host.Tables.Config;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+namespace Microsoft.Extensions.Hosting
+{
+    public static class StorageWebJobsBuilderExtensions
```
Comment (christothes) [inline sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/StorageWebJobsBuilderExtensions.cs:12]:
```suggestion
    public static class TablesWebJobsBuilderExtensions
```

File: sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/StorageWebJobsBuilderExtensions.cs (hunk 4 above)
Comment (christothes) [inline sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/StorageWebJobsBuilderExtensions.cs:14]:
```suggestion
        public static IWebJobsBuilder AddAzureTables(this IWebJobsBuilder builder)
```

**Review by christothes — APPROVED** [review]

**Comment by pakrym** [comment]
>Is the idea here that we are trying to merge in what would be the T1 version of the extension but using the package naming pattern established in T2?

No, we are going to turn this into a full T2 extension. As a first step, I want to migrate existing code as is so it's easier to review future changes.

**Comment by pakrym** [comment]
@christothes I agree with all of your suggestions. They are going to follow in the next PR.

**Pull request merged**